### PR TITLE
cmd/devp2p/internal/ethtest: only use eth66 if eth66 is negotiated

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
Fixes an issue where we send 65 style messages over a 66 connection

### CODE REVIEW

File: cmd/devp2p/internal/ethtest/helpers.go [hunk 1]
```diff
@@ -242,18 +242,25 @@ func (s *Suite) createSendAndRecvConns(isEth66 bool) (*Conn, *Conn, error) {
...
 // readAndServe serves GetBlockHeaders requests while waiting
 // on another message from the node.
-func (c *Conn) readAndServe(chain *Chain, timeout time.Duration) Message {
+func (c *Conn) readAndServe65(chain *Chain, timeout time.Duration) Message {
 	start := time.Now()
 	for time.Since(start) < timeout {
 		c.SetReadDeadline(time.Now().Add(5 * time.Second))
 		switch msg := c.Read().(type) {
-		case *Ping:
+		case Ping:
```
Comment (holiman) [inline cmd/devp2p/internal/ethtest/helpers.go:260]: This change looks like it either makes something that didn't work, work, or vice versa. What's that about?

Comment (MariusVanDerWijden) [reply cmd/devp2p/internal/ethtest/helpers.go:260]: Ha, yes it fixes something, but I accidentally fixed it in the wrong method

Comment (MariusVanDerWijden) [reply cmd/devp2p/internal/ethtest/helpers.go]: Read66 returns `GetBlockHeaders` instead of `*GetBlockHeaders`which is returned by 65

**Review by holiman — APPROVED** [review]
LGTM

**Pull request merged**